src/Tables/atoms/TableHeaderTitle.tsx

```diff
@@ -0,0 +1,83 @@
+import React from 'react';
+import styled, {css}from 'styled-components';
+import { ITableColumnConfig } from '..';
+import Icon, {IconWrapper} from '../../Icons/Icon';
+
+const HeaderTitle = styled.div<{sortable?: boolean, isSortActive?: boolean, ascending?:boolean}>`
+  border-bottom-right-radius: 3px;
+  border-top-right-radius: 3px;
+  display: inline-flex;
+  justify-content: center;
+  align-content: center;
+  position: relative;
+  padding: 0 2px;
+  user-select: none;
+
+  ${IconWrapper} {
+    position: absolute;
+    top: 0px;
+    left: -15px;
+    display: none;
+    border-bottom-left-radius: 3px;
+    border-top-left-radius: 3px;
+    padding: 3px 1px;
+    
+    ${({ascending}) => ascending && css`
+      transform: rotate(180deg);
+      border-bottom-right-radius: 3px;
+      border-top-right-radius: 3px;
+      border-bottom-left-radius: 0;
+      border-top-left-radius: 0;
+      top: 0;
+    `}
+  }
+  
+  ${({isSortActive}) => isSortActive && css`
+    &:not(hover) {
+      ${IconWrapper} {
+        display: inline-flex;
+      }  
+    }
+  `}
+
+  ${({sortable}) => sortable && css`
+    &:hover {
+      background-color: hsl(220, 12%, 95%);
+      cursor: pointer;
+      ${IconWrapper} {
+        display: inline-flex;
+        background-color: hsl(220, 12%, 95%);
+      }
+    }
+  `}
+`;
+
+interface IHeaderProps extends ITableColumnConfig {
+  isSortActive?: boolean
+  ascending?: boolean
+  columnKey: number
+  toggleSort: (columnKey: number) => void
+}
+
+const TableHeaderTitle : React.FC<IHeaderProps> = ({
+  header,
+  sortable,
+  isSortActive,
+  ascending,
+  columnKey,
+  toggleSort
+}) => {
+  return(
+      <HeaderTitle
+        sortable={sortable}
+        isSortActive={isSortActive}
+        ascending={ascending}
+        onClick= {() => {toggleSort(columnKey)}}
```

Comment:
Minor: should this be a useCallback.

Reply:
I might have misunderstood the question but this toggleSort is on a useCallback.

toggleSort here is a executed prop. This function is wrapped in a useCallback handled by the parent component TypeTable.

<img width="1468" alt="Screen Shot 2021-03-26 at 16 53 16 1" src="https://user-images.githubusercontent.com/10409078/112600075-0eb3e600-8e54-11eb-9b2b-f9e0e8e5e715.png">

Reply:
there is a implicit function declaration there. that is all i meant.

Reply:
Yes this is an implicit declaration I will update it to explicit if you prefer.
However I will not add a useCallback to it because that creates a very weird behavior of only re-rendering only if another HeaderTitle is sorted.